### DISCUSSION TITLE
Passing a Javascript result to Livewire? - Unlayer Editor

### DISCUSSION THREAD

**Discussion by MCKLtech** [livewire/livewire#5316] [1. Help] [answered]
I'm using Unlayer as an editor in a project, and I want to export the layout to Livewire so I can persist it in the database. The Javascript method looks like this:

```
unlayer.exportHtml(function(data) {
  var json = data.design; // design json
  var html = data.html; // final html

  // Do something with the json and html
})
```
Ref: https://docs.unlayer.com/docs/export-html

The question is how would I pass the json and HTML variables to Livewire? Livewire handles the Save of the overall page, so my guess is I need to trigger the exportHTML with Livewire during that cycle, grab the variables, and save them. The question is, how?

Or, is there a better way?

**Comment by hamzaelmaghari** [comment, marked as answer]
## Hey @MCKLtech !
Check this out, it might be helpfull 🤞

[Checkout](https://laravel-livewire.com/docs/2.x/events)

**Reply by MCKLtech** [reply]
Thanks @hamzaelmaghari , I actually solved this with events.

In PHP, when saved, I do an emit:

```
public function saveMe() {

//Save stuff

$this->emit('fetchUnlayer');
}
```

And on the Livewire side, I emit it back:

```
<script>
                Livewire.on('fetchUnlayer', postId => {
                    unlayer.exportHtml(function (data) {
                        Livewire.emit('updateUnlayerHTML', data.html);
                        Livewire.emit('updateUnlayerJSON', data.design);
                    }, {
                        minify: true
                    })
                })
            </script>
```

Then, on the PHP side, I have functions that take the data and process it:

```
public function updateUnlayerHTML($html) {
 //Do Stuff
}
```
